wordpress-mobile/WordPress-Android
# Issue/8482 privacy settings crash

### Fix
Add the _**Crash reports**_ item to _**Privacy settings**_ similar to the [WooCommerce crash reports privacy settings feature](https://github.com/woocommerce/woocommerce-android/pull/443) as described in https://github.com/wordpress-mobile/WordPress-Android/issues/8482.  See the screenshots below for illustration.

![privacy_settings](https://user-images.githubusercontent.com/3827611/50450502-89e86380-08fc-11e9-8caa-680a95ad560d.png)

#### Note
The WordPress implementation differs from the WooCommerce app in a few ways.  The default background was used rather than the card background as shown in the WooCommerce app.  The preference dividers were left as full-width rather than shortening them as shown in the WooCommerce app.  Both of those were excluded from these changes so that _**Privacy settings**_ would remain consistent with the rest of the app

### Test
Crashlytics are not enabled for debug builds so these changes must be tested on a release build or [this `if` statement](https://github.com/wordpress-mobile/WordPress-Android/blob/d0200999a091f3d3c6e0ed0bbfc39975092bae3f/WordPress/src/main/java/org/wordpress/android/util/CrashlyticsUtils.java#L19-L21) must be commented out while running a debug build.  The steps below assume the latter.

0. Set breakpoint [here](https://github.com/wordpress-mobile/WordPress-Android/blob/d0200999a091f3d3c6e0ed0bbfc39975092bae3f/WordPress/src/main/java/org/wordpress/android/util/CrashlyticsUtils.java#L25).
1. Launch WordPress app.
2. Notice `hasUserOptedOut` is `false`.
3. Go to ***Me*** tab.
4. Tap ***App Settings*** item.
5. Tap ***Privacy Settings*** item.
6. Notice updated layout as shown in screenshots above.
7. Notice _**Crash reports**_ item is enabled.
8. Tap _**Crash reports**_ item to disabled it.
9. Exit WordPress app.
10. Launch WordPress app.
11. Notice `hasUserOptedOut` is `true`.
12. Go to ***Me*** tab.
13. Tap ***App Settings*** item.
14. Tap ***Privacy Settings*** item.
15. Notice _**Crash reports**_ item is disabled.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
The [`RELEASE-NOTES.txt`](https://github.com/wordpress-mobile/WordPress-Android/blob/d0200999a091f3d3c6e0ed0bbfc39975092bae3f/RELEASE-NOTES.txt) document was not updated since the release with the [original feature pull request](https://github.com/wordpress-mobile/WordPress-Android/pull/7768) did not include notes.  However, this pull request does include user-facing changes.  I'll update the release notes if the reviewers see fit.
